test/images/agnhost/connect/connect.go

```diff
@@ -125,3 +132,54 @@ func connectSCTP(dest string, timeout time.Duration) {
 		os.Exit(1)
 	}
 }
+
+func connectUDP(dest string, timeout time.Duration, data string) {
+	var (
+		readBytes int
+		buf       = make([]byte, 1024)
+	)
+
+	if _, err := net.ResolveUDPAddr("udp", dest); err != nil {
+		fmt.Fprintf(os.Stderr, "DNS: %v\n", err)
+		os.Exit(1)
+	}
+
+	conn, err := net.Dial("udp", dest)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "OTHER: %v\n", err)
+		os.Exit(1)
+	}
+
+	if timeout > 0 {
+		if err = conn.SetDeadline(time.Now().Add(timeout)); err != nil {
+			fmt.Fprintf(os.Stderr, "OTHER: %v\n", err)
+			os.Exit(1)
+		}
+	}
+
+	if _, err = conn.Write([]byte(fmt.Sprintf("%s\n", data))); err != nil {
+		parseUDPErrorAndExit(err)
+	}
+
+	if readBytes, err = conn.Read(buf); err != nil {
+		parseUDPErrorAndExit(err)
+	}
+
+	// ensure the response from UDP server
+	if readBytes == 0 {
```

Comment:
So, basically, we just have to make sure that the UDP server sends us some data back, the content of which is unimportant, right?

Reply:
I think this is the more general approach .... we can complicate it more and add an option to match the answer

Reply:
more history about this https://nmap.org/book/scan-methods-udp-scan.html#:~:text=Fortunately%2C%20Nmap%20can%20help%20inventory,packet%20to%20every%20targeted%20port.